ARCHITECTURE.md

```diff
@@ -0,0 +1,57 @@
+# Rules Rust - Architecture
+
+In this file we describe how we think about the architecture
+of `rules_rust`. It's goal is to help contributors orient themselves, and to
+document code restrictions and assumptions.
+
+In general we try to follow the common standard defined by
+https://docs.bazel.build/versions/master/skylark/deploying.html.
+
+## //rust
+
+This is the core package of the rules. It contains all the core rules such as
+`rust_binary` and `rust_library`. It also contains `rust_common`, a Starlark
+struct providing all rules_rust APIs that one might need when writing custom
+rules integrating with rules_rust.
+
+`//rust` and all its subpackages have to be standalone. Users who only need
+the core rules should be able to ignore all other packages.
+
+`//rust:defs.bzl` is the file that all users of `rules_rust` will be using.
+Everything in this file can be used (dependend on) and is supported. Typically
```

Comment:
This somewhat raises the question of compatibility guarantees - "can be used and is supported"... Until when?

Reply:
That's a hard question which I'd like to find answer to in a separate discussion. Would you prefer uncertainty of the current text, or deleting it?

Reply:
I'd suggest either adding an explicit qualifier ("though stability is not currently guaranteed across commits", and a link to an issue), or removing it.

Reply:
Good point. Done.